src/Domain/Creators/BinaryCompactObjectHelpers.cpp

```diff
@@ -129,76 +129,71 @@ void TimeDependentMapOptions::build_maps(
-TimeDependentMapOptions::frame_to_inertial_map() const {
-  return std::make_unique<
-      CubicScaleAndRotationMapForComposition<SourceFrame, Frame::Inertial>>(
-      expansion_map_, rotation_map_);
+TimeDependentMapOptions::MapType<Frame::Distorted, Frame::Inertial>
+TimeDependentMapOptions::distorted_to_inertial_map(
+    const bool include_distorted_map) const {
+  if (include_distorted_map) {
+    return std::make_unique<DistortedToInertialComposition>(expansion_map_,
+                                                            rotation_map_);
+  } else {
+    return nullptr;
+  }
 }
 
 template <domain::ObjectLabel Object>
 TimeDependentMapOptions::MapType<Frame::Grid, Frame::Distorted>
-TimeDependentMapOptions::grid_to_distorted_map(const bool use_identity) const {
-  if (use_identity) {
-    return std::make_unique<
-        IdentityForComposition<Frame::Grid, Frame::Distorted>>(IdentityMap{});
-  } else {
+TimeDependentMapOptions::grid_to_distorted_map(
+    const bool include_distorted_map) const {
+  if (include_distorted_map) {
     const size_t index = get_index<Object>();
-    return std::make_unique<
-        CompressionMapForComposition<Frame::Grid, Frame::Distorted>>(
+    return std::make_unique<GridToDistortedComposition>(
         gsl::at(size_maps_, index));
+  } else {
+    return nullptr;
   }
 }
 
 template <domain::ObjectLabel Object>
 TimeDependentMapOptions::MapType<Frame::Grid, Frame::Inertial>
-TimeDependentMapOptions::everything_grid_to_inertial_map(
+TimeDependentMapOptions::grid_to_inertial_map(
     const bool include_distorted_map) const {
   if (include_distorted_map) {
     const size_t index = get_index<Object>();
-    return std::make_unique<EverythingMapForComposition>(
+    return std::make_unique<GridToInertialComposition<true>>(
         gsl::at(size_maps_, index), expansion_map_, rotation_map_);
   } else {
-    return std::make_unique<EverythingMapNoDistortedForComposition>(
-        IdentityMap{}, expansion_map_, rotation_map_);
+    return std::make_unique<GridToInertialComposition<false>>(expansion_map_,
+                                                              rotation_map_);
   }
 }
 
 template <domain::ObjectLabel Object>
 size_t TimeDependentMapOptions::get_index() const {
-  ASSERT(Object == domain::ObjectLabel::A or Object == domain::ObjectLabel::B,
-         "Object label for TimeDependentMapOptions must be either A or B, not"
+  ASSERT(Object == domain::ObjectLabel::A or Object == domain::ObjectLabel::B or
+             Object == domain::ObjectLabel::None,
+         "Object label for TimeDependentMapOptions must be either A, B or "
+         "None, not"
              << Object);
-  return Object == domain::ObjectLabel::A ? 0_st : 1_st;
+  return Object == domain::ObjectLabel::A
+             ? 0_st
+             : (Object == domain::ObjectLabel::B
+                    ? 1_st
+                    : std::numeric_limits<size_t>::max());
```

Comment:
You don't/shouldn't call this function with `ObjectLabel::None`, right? Then keep the assert and the condition here unchanged.

Reply:
Yeah I guess because I just `ASSERT` and don't `static_assert` that would work.